api/routes/qaris.js

```diff
@@ -42,4 +44,68 @@ router.get('/related/:id', (req, res) => {
     .then(related => res.send(related));
 });
 
+router.get('/:id/download', (req, res) => {
+  const archive = archiver('zip', {
+    zlib: { level: 9 } // Sets the compression level.
+  });
+  const header = {
+    'Content-Type': 'application/x-zip',
+    Pragma: 'public',
+    Expires: '0',
+    'Cache-Control': 'private, must-revalidate, post-check=0, pre-check=0',
+    'Content-disposition': `attachment; filename=\"complete_quran.zip\"`,
+    'Transfer-Encoding': 'chunked',
+    'Content-Transfer-Encoding': 'binary'
+  };
+
+  archive
+    .on('warning', err => {
+      if (err.code === 'ENOENT') {
+        console.log(err);
+      } else {
+        throw err;
+      }
+    })
+    .on('error', err => {
+      throw err;
+    })
+    .on('finish', () => res.send());
```

Comment:
We should send something for the frontend?

Reply:
You mean in addition to the pipped zip file?

Reply:
I think it means usually with `res.send` or `res.end` you'd pass some text or content. But I think in this case because of the at this point (finish) the zip is delivered and the `res.send`  is called back  `archive.finalize();`. Maybe use: `res.status(200).end();` ?

Reply:
https://github.com/archiverjs/node-archiver/blob/master/examples/express.js

based on the example above you don't need `.on('finish', () => res.send());`